Option for current tab number to secondary section

macownersclub opened the pull request:
# option:
The number of the current tab is displayed to the right (secondary section) in the form of `-1-`.
This will happen only if:
- tab-bar-mode is enabled
- the custom variable nano-modeline-display-tab-number is set to `t` and
- there are more than 1 tab open

# changes to files:
- nano-modeline.el:
  - Add custom variable nano-modeline-display-tab-number
  - Add function nano-modeline-tab-number
  - Modify function nano-modeline-render

rougier commented:
Thanks. Could you post a screenshot of the result?

macownersclub commented:
Bonjour,

Of course I can.

This ist the mode-line without the number:

<img width="1482" alt="nano-mode-line-without-tab-number" src="https://user-images.githubusercontent.com/84017200/211598388-36256454-012f-4477-ac24-e5c758989ba8.png">


This is with the number:

<img width="1497" alt="nano-mode-line-with-tab-number" src="https://user-images.githubusercontent.com/84017200/211598457-b75d5cd4-c9df-4689-aa88-cad97efffa1c.png">

rougier commented on nano-modeline.el:
```diff
@@ -483,7 +502,12 @@ KEY mode name, for reference only. Easier to do lookups and/or replacements.
+                                  'display `(raise ,nano-modeline-space-bottom))
+                 (if nano-modeline-display-tab-number
+                     (if (not (equal "0" (nano-modeline-tab-number)))
+                    (concat (propertize " -" 'face face-secondary)
```
You could use `(propertize (format " -%d- " nano-modeline-tab-number) 'face face-secondary)` and/or have a custom option to specify the format.

rougier commented:
Thanks. Did you sign the [Emacs copyright assignment](https://www.gnu.org/software/emacs/manual/html_node/emacs/Copyright-Assignment.html)? (I cannot merge else)

Also see my comment on custom format option

macownersclub commented:
Updated and rebased your suggestions, since you are right.
The copyright assignment is gonna take a "minute", but I am on it since I want to try and contribute a bit more if you don't mind.

rougier commented:
Sure. Tell me when it is signed.

macownersclub commented:
Hello there,

Just got this email from the fsf:

> Your assignment/disclaimer process with the FSF is currently
> complete; your fully executed PDF will be sent to you in a separate
> email immediately following this one.

rougier commented:
And many thanks for your contribution!
